client.go

```diff
@@ -240,11 +246,19 @@ func NewClientRaw(
 	if len(cfg.Network.URLs) > 1 {
 		L.Warn().Msg("Multiple RPC URLs provided, only the first one will be used")
 	}
-
-	client, err := ethclient.Dial(cfg.Network.URLs[0])
+	ctx, cancel := context.WithTimeout(context.Background(), cfg.Network.DialTimeout.Duration())
+	defer cancel()
+	rpcClient, err := rpc.DialOptions(ctx,
+		cfg.Network.URLs[0],
+		rpc.WithHeaders(cfg.RPCHeaders),
+		rpc.WithHTTPClient(&http.Client{
+			Transport: NewLoggingTransport(),
+		}),
+	)
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to '%s' due to: %w", cfg.Network.URLs[0], err)
+		return nil, fmt.Errorf("failed to connect RPC client to '%s' due to: %w", cfg.Network.URLs[0], err)
```

Comment:
nit: would be nice to have a helper method on `Config` to get `cfg.Network.URLs[0]` just like we have for root address

Reply:
done